### DISCUSSION TITLE
Quarkus AI - getting started

### DISCUSSION THREAD

**Discussion by donbourne** [quarkusio/quarkus#44205] [Community]
Hello,

I've been looking into how best to build new Java-based AI apps, and in particular investigating the Quarkus way vs. the Spring AI way. While both provide capabilities to interact with LLMs for generating responses, streaming, structuring input/output, using tools, etc. they do so in different ways. I wanted to share some of my observations here, and see which ones people think are worth addressing and to get ideas to address them. My hope is that we can improve the user experience of the Quarkus AI solution.

For this discussion, I'll focus on the first steps people need to take to find info and learn about the Quarkus AI solution. I can post more discussion topics separately on other aspects if people think this is worthwhile.

What I tried...
- I Googled for `Quarkus AI`, and read the info I found there, then tried to run an example from the Ollama page.

Challenges...
1. When Googling for `Quarkus AI`, the first two links are random blogs. The third link (https://docs.quarkiverse.io/quarkus-langchain4j/dev/) is more useful (see image below). It would be good if the SEO could be worked on to get the third link up to the top. In contrast, Spring AI's reference doc is the first link returned.

![image](https://github.com/user-attachments/assets/92ed803b-4a23-4408-bc0c-584db19f9ab9)

Google results for `Quarkus AI`

![image](https://github.com/user-attachments/assets/47c41172-904f-4830-bd08-207b92ee7ca7)

Google results for `Spring AI`

2. regarding the Quarkus LangChain4J getting started guide (from the third link I found above):
- it seems to assume the reader is familiar with LangChain4J.  It would be better if it could explain briefly what LC4J is and that it's different than just a port of LangChain.
- it seems to assume the reader is familiar with Gen AI concepts. In contrast, the Spring AI doc at https://docs.spring.io/spring-ai/reference/ gives a good intro to AI Concepts at https://docs.spring.io/spring-ai/reference/concepts.html. This may seem unnecessary, but I'm more likely to use the thing that teaches me the basics, all other things being equal.
- it doesn't give a comprehensive list of capabilities provided by the Quarkus LangChain4J extension. Ideally that would exist and include everything from LangChain4J and from the add ons in the extension. It would be good if each capability was backed up by a link to documentation that gives details on that capability and how to use it.
- it appears to compete with vanilla LangChain4J. Currently it presents a list of ways in which Quarkus LangChain4J is better than vanilla LangChain4J. IMO it would be better to just describe how it augments vanilla LangChain4J, without appearing to be competing with vanilla LangChain4J.
- it highlights one capability. It currently provides a one line section about EasyRAG, but nothing about other kinds of LLM interactions. This is a bit confusing. As mentioned previously, I'd suggest mentioning all of the capabilities of the combined LC4J and Quarkus LC4J.

The most important point may be that Quarkus LangChain4J docs currently seem like a sparse add on on top of an assumed knowledge of LangChain4J. Spring AI hasn't had to grapple with how to document things in the same way since it provides its own self-contained API -- and the result is that the Spring AI doc ends up looking more complete and easier to follow.

Does this seem like a good set of changes?

**Comment by jmartisk** [comment]
Hi @donbourne, that's a lot of good points that I agree with. Any PRs with improvements to the quarkus-langchain4j docs are appreciated (the docs reside in https://github.com/quarkiverse/quarkus-langchain4j/tree/main/docs) :)

**Reply by donbourne** [reply]
Thanks @jmartisk , I'll see what I can do on that! :smile:
